File: Easy/3375. Minimum Operations to Make Array Values Equal to K/Solution.cpp
class Solution {
public:
    int minOperations(vector<int> &nums, int k) {
        unordered_set<int> distinctNums;
        distinctNums.emplace(k);

        int minNum = nums[0];

        for (int i = 0; i < (int) nums.size(); i++) {
            minNum = min(minNum, nums[i]);
            distinctNums.emplace(nums[i]);
        }

        if (k > minNum) return -1;
        return (int)distinctNums.size() - 1;
    }
};

File: Easy/3375. Minimum Operations to Make Array Values Equal to K/Solution.py
class Solution(object):
    def minOperations(self, nums, k):
        distinctNums = set()
        distinctNums.add(k)

        minNum = nums[0]

        for i in range(len(nums)):
            minNum = min(minNum, nums[i])
            distinctNums.add(nums[i])

        if k > minNum:
            return -1
        
        return len(distinctNums) - 1
